src/preamble.js

```diff
@@ -1822,13 +1822,11 @@ var runDependencyTracking = {};
 #endif
 
 function getUniqueRunDependency(id) {
-#if ASSERTIONS
   var orig = id;
   while (1) {
     if (!runDependencyTracking[id]) return id;
```

Comment:
`runDependencyTracking` will only exist is `ASSERTIONS`.

Did you run into a bug where run deps with duplicate names were added, and this led to a bug? The ids should only be used for debugging (in `ASSERTIONS`). The actually important thing is just the `runDependencies` counter.